src/test/java/com/artipie/nuget/NugetITCase.java

```diff
@@ -23,120 +23,43 @@
-import org.hamcrest.MatcherAssert;
+import org.hamcrest.core.IsEqual;
 import org.hamcrest.core.StringContains;
 import org.hamcrest.text.StringContainsInOrder;
-import org.junit.jupiter.api.AfterEach;
 import org.junit.jupiter.api.BeforeEach;
-import org.junit.jupiter.api.Disabled;
 import org.junit.jupiter.api.Test;
-import org.junit.jupiter.api.Timeout;
 import org.junit.jupiter.api.condition.EnabledOnOs;
 import org.junit.jupiter.api.condition.OS;
-import org.testcontainers.Testcontainers;
-import org.testcontainers.shaded.org.apache.commons.io.FileUtils;
+import org.junit.jupiter.api.extension.RegisterExtension;
 
 /**
  * Integration tests for Nuget repository.
  * @since 0.12
- * @checkstyle ClassDataAbstractionCouplingCheck (500 lines)
- * @checkstyle MagicNumberCheck (500 lines)
  */
 @SuppressWarnings("PMD.AvoidDuplicateLiterals")
 @EnabledOnOs({OS.LINUX, OS.MAC})
 final class NugetITCase {
 
     /**
-     * URL.
+     * Test deployments.
+     * @checkstyle VisibilityModifierCheck (10 lines)
      */
-    private static final String URL = "http://host.testcontainers.internal:%d/my-nuget/index.json";
-
-    /**
-     * Temporary directory for all tests.
-     */
-    private Path tmp;
-
-    /**
-     * Tested Artipie server.
-     */
-    private ArtipieServer server;
-
-    /**
-     * Container.
-     */
-    private TestContainer cntn;
-
-    /**
-     * Server port.
-     */
-    private int port;
+    @RegisterExtension
+    final TestDeployment containers = new TestDeployment(
+        () -> TestDeployment.ArtipieContainer.defaultDefinition()
+            .withRepoConfig("nuget/nuget.yml", "my-nuget"),
+        () -> new TestDeployment.ClientContainer("mcr.microsoft.com/dotnet/sdk:5.0")
+            .withWorkingDirectory("/w")
+    );
 
     @BeforeEach
-    void init() throws Exception {
-        this.tmp = Files.createTempDirectory("junit-nuget");
-        final String name = "my-nuget";
-        this.port = new RandomFreePort().value();
-        this.server = new ArtipieServer(this.tmp, name, this.config().toString(), this.port);
-        this.server.start();
-        Testcontainers.exposeHostPorts(this.port);
-        this.createNugetConfig();
-        this.cntn = new TestContainer("mcr.microsoft.com/dotnet/sdk:5.0", this.tmp);
-        this.cntn.start(this.port);
-    }
-
-    @AfterEach
-    @SuppressWarnings("PMD.AvoidPrintStackTrace")
-    void tearDown() {
-        this.server.stop();
-        this.cntn.close();
-        try {
-            FileUtils.cleanDirectory(this.tmp.toFile());
-        } catch (final IOException ex) {
-            ex.printStackTrace();
-        }
-    }
-
-    @Test
-    @Timeout(10)
-    void shouldPushPackage() throws Exception {
-        MatcherAssert.assertThat(
-            this.pushPackage(),
-            new StringContains("Your package was pushed.")
-        );
-    }
-
-    @Test
-    @Timeout(30)
-    @Disabled
-    void shouldInstallPushedPackage() throws Exception {
```

Comment:
@olenagerasimova also one concern about this test case. Probably remain `@todo` to add installation test or rewrite it